lib/join_collection.rb

```diff
@@ -82,7 +86,7 @@ def join_data(mapper, key, delegate_if, delegate_fields)
         when plural_target
           doc[plural_target] = target_objects if join_type == :join_many
         else
-          doc["#{singular_target}_#{field}"] = target_object.try(field)
+          doc["#{singular_target}_#{field}"] = target_object.try!(field)
```

Comment:
Why do you want to throw out an exception here? I think if the field doesn't exist, we will have a `nil` value so the program can keep going.